File: java/src/main/java/ai/buble/sdk/generations/GenerationResult.java
package ai.buble.sdk.generations;

import java.util.List;

/**
 * Generated media assets returned by a successful generation task.
 */
public class GenerationResult {
    private List<MediaResultImage> images;
    private List<MediaResultVideo> videos;
    private List<MediaResultAudio> audios;

    public List<MediaResultImage> getImages() { return images; }
    public void setImages(List<MediaResultImage> images) { this.images = images; }
    public List<MediaResultVideo> getVideos() { return videos; }
    public void setVideos(List<MediaResultVideo> videos) { this.videos = videos; }
    public List<MediaResultAudio> getAudios() { return audios; }
    public void setAudios(List<MediaResultAudio> audios) { this.audios = audios; }
}

File: dotnet/src/Buble.Sdk/Generations/GenerationResult.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace Buble.Sdk.Generations;

public sealed class GenerationResult
{
    [JsonPropertyName("images")]
    public IReadOnlyList<MediaResultImage>? Images { get; set; }

    [JsonPropertyName("videos")]
    public IReadOnlyList<MediaResultVideo>? Videos { get; set; }

    [JsonPropertyName("audios")]
    public IReadOnlyList<MediaResultAudio>? Audios { get; set; }

    [JsonExtensionData]
    public Dictionary<string, JsonElement>? AdditionalProperties { get; set; }
}
